Fix the replica should remove the master link when receiving the CLUSTER RESET command

git-hulk opened the pull request:
This PR also allows using SOFT|HARD in the CLUSTER RESET command,
but it has the same meaning in the Kvrocks cluster since keeping
the cluster's current epoch(version) is meaningless to it.

git-hulk commented on tests/gocase/integration/cluster/cluster_test.go:
```diff
@@ -473,33 +473,51 @@ func TestClusterReset(t *testing.T) {
+		require.Contains(t, rdb0.ClusterResetHard(ctx).Err(), "Can't reset cluster while database is not empty")
 		require.NoError(t, rdb0.Del(ctx, key).Err())
-		require.NoError(t, rdb0.Do(ctx, "cluster", "reset").Err())
+		require.NoError(t, rdb0.ClusterResetSoft(ctx).Err())
```
Use to confirm the `CLUSTER RESET SOFT` also works well.

PragmaTwice commented on src/storage/rdb.cc:
```diff
@@ -125,7 +125,7 @@ StatusOr<uint64_t> RDB::loadObjectLen(bool *is_encoded) {
         return ntohu64(len);
       } else {
-        return {Status::NotOK, fmt::format("Unknown RDB string encoding type {} byte {}", type, c)};
+        return {Status::NotOK, fmt::format("Unknown RDB co -string encoding type {} byte {}", type, c)};
```
Is there any special meaning of `co -`?

git-hulk replied:
No, I wrongly typed the keyboard and submit the change.

git-hulk replied:
Fixed now.

git-hulk committed:
Fix the replica should remove the master link when receiving the CLUSTER RESET command

This PR also allows using SOFT|HARD in the CLUSTER RESET command, but
it's the same meaning to the Kvrocks cluster since kept the cluster
current epoch(version) is meaningless to it.

git-hulk committed:
Merge branch 'unstable' into fix-remove-master-in-cluster-reset
